vllm/engine/async_llm_engine.py

```diff
@@ -161,12 +169,15 @@ def get_new_and_finished_requests(self) -> Tuple[List[Dict], Set[str]]:
             self._request_streams[stream.request_id] = stream
             new_requests.append(new_request)
 
-        self.new_requests_event.clear()
-
         return new_requests, finished_requests
 
     async def wait_for_new_requests(self):
-        await self.new_requests_event.wait()
+        if not self.has_new_requests():
+            await self.new_requests_event.wait()
+        self.new_requests_event.clear()
```

Comment:
Suggestion to only clear before waiting
```suggestion
        if not self.has_new_requests():
            self.new_requests_event.clear()
            if not self.has_new_requests():
                await self.new_requests_event.wait()
```

Reply:
Hmm can you explain why we should do it like that?

Reply:
Yeah I think it should be fine